pallets/investments/src/lib.rs

```diff
@@ -708,190 +634,218 @@ where
-				if order.submitted_at() == cur_order_id {
-					Self::deposit_event(Event::<T>::RedeemCollectedForNonClearedOrderId {
-						who: who.clone(),
-						investment_id,
-					});
-					// TODO: Return correct weight
-					//       - Accountant::info() + 2 * Storage::read() + Storage::write()
-					return Ok(().into());
-				}
-
-				for order_id in order.submitted_at()..last_processed_order_id {
-					let fulfillment = ClearedRedeemOrders::<T>::try_get(investment_id, order_id)
-						.map_err(|_| Error::<T>::OrderNotCleared)?;
-					Pallet::<T>::acc_payout_redeem(&mut collection, &fulfillment)?;
-					Pallet::<T>::acc_remaining_redeem(&mut collection, &fulfillment)?;
-					collected_ids.push(order_id);
-				}
-
-				order.update_after_collect(
-					collection.remaining_investment_redeem,
-					last_processed_order_id,
-				);
+		RedeemOrders::<T>::try_mutate(&who, investment_id, |maybe_order| {
+			// Exit early if order does not exist
+			let order = if let Some(order) = maybe_order.as_mut() {
+				order
+			} else {
+				// Trigger event
+				Self::deposit_event(Event::<T>::RedeemCollectedWithoutActivePosition {
+					who: who.clone(),
+					investment_id,
+				});
+				// TODO: Return correct weight
+				//       - Accountant::info() + Storage::read() + Storage::write()
+				return Ok((Default::default(), ().into()));
+			};
+
+			let mut collection = RedeemCollection::<T::Amount>::from_order(order);
+			let mut collected_ids = Vec::new();
+			let cur_order_id = RedeemOrderId::<T>::get(investment_id);
+			let last_processed_order_id = min(
+				order
+					.submitted_at()
+					.saturating_add(T::MaxOutstandingCollects::get()),
+				cur_order_id,
+			);
+
+			// Exit early if the current order is not in processing
+			if order.submitted_at() == cur_order_id {
+				Self::deposit_event(Event::<T>::RedeemCollectedForNonClearedOrderId {
+					who: who.clone(),
+					investment_id,
+				});
+				// TODO: Return correct weight
+				//       - Accountant::info() + 2 * Storage::read() + Storage::write()
+				return Ok((Default::default(), ().into()));
+			}
 
-				// Transfer collected amounts from investment and redemption
-				let investment_account =
-					InvestmentAccount { investment_id }.into_account_truncating();
-				T::Tokens::transfer(
-					info.payment_currency(),
-					&investment_account,
-					&who,
-					collection.payout_investment_redeem,
-					false,
+			let mut amount_payment: T::Amount = Default::default();
+			for order_id in order.submitted_at()..last_processed_order_id {
+				let fulfillment = ClearedRedeemOrders::<T>::try_get(investment_id, order_id)
+					.map_err(|_| Error::<T>::OrderNotCleared)?;
+				let payout_tranche_tokens =
+					Pallet::<T>::acc_payout_redeem(&mut collection, &fulfillment)?;
+				Pallet::<T>::acc_remaining_redeem(&mut collection, &fulfillment)?;
+				collected_ids.push(order_id);
+
+				// TODO(@mustermeiszer): We actually want the reciprocal without rounding, is
```

Comment:
NOTE: Similar to above `do_collect_invest`. Here, we determine the amount of tranche tokens which were converted into pool currency.

Q: See inline comment
>  We actually want the reciprocal without rounding, is this sufficient or should we use something like `reciprocal_with_rounding(SignedRounding::NearestPrefMajor)`?